### DISCUSSION TITLE
Are there any plans for a GTK 4 port?

### DISCUSSION THREAD

**Discussion by Newbytee** [SoongNoonien/mpdevil#78] [Q&A] [answered]
Are there any plans to port mpdevil to GTK 4? If so, is this something you are working on, or would you be open to outside contributions implementing this? I see that there's a branch in this repo, but it doesn't seem to have been updated for over two years: https://github.com/SoongNoonien/mpdevil/commits/gtk4

**Comment by SoongNoonien** [comment, marked as answer]
> Are there any plans to port mpdevil to GTK 4?

Yes, there are but it may still take a while.

> If so, is this something you are working on, or would you be open to outside contributions implementing this?

It is still in the planing phase as I'm thinking of perhaps adopting libadwaita. Therefore I most likely wouldn't except pull requests regarding such big changes.

> I see that there's a branch in this repo, but it doesn't seem to have been updated for over two years: https://github.com/SoongNoonien/mpdevil/commits/gtk4

This was a test but gtk4 was not very usable back then.

**Comment by SoongNoonien** [comment]
A GTK4 port is now in master. Obviously it is still a bit broken but I'm working on it.

**Reply by ghost** [reply]
Hi
I'm using latest git of mpdevil and love the work you are doing on the GTK4 port!

One thing I have noticed is that when doing a search by "Artist" or "AlbumArtist" search returns individual songs rather than just artists. I have attached a screenshot.
Appreciate the port is a work in progress but thought I'd just highlight it here.

Thanks for all your work on this project.

Richard

![image](https://github.com/SoongNoonien/mpdevil/assets/117611987/33a587fb-150d-4432-8a05-e5ac9e6fc9a4)

**Reply by SoongNoonien** [reply]
Hi!
This is actually the intended behavior and is not specific to the GTK4 port. The search view is to search for individual songs only. Choosing "AlbumArtist" on the left simply has the effect that only songs are shown where the "AlbumArtist" tag contains the search request. It is planned to support a search/filter functionality for artists in the future. Take a look at https://github.com/SoongNoonien/mpdevil/issues/86 for reference.

**Comment by ghost** [comment]
Thanks for the reply!
Ah, that makes sense, I had incorrectly thought that the main page
Artist filter had been dropped in favour of the Search feature.

I'm very much looking forward to the first stable GTK4 release.

My bad for not noting the issue had already been raised on the issue
tracker but, on a side note, would you prefer comments for "Master" to
be raised on the issue tracker or kept separate in discussions /
emailed ?

Richard


-----Original Message-----
From: Martin Wagner ***@***.***>
Reply-To: SoongNoonien/mpdevil
***@***.***>
To: SoongNoonien/mpdevil ***@***.***>
Cc: Richjn ***@***.***>, Comment ***@***.***>
Subject: Re: [SoongNoonien/mpdevil] Are there any plans for a GTK 4
port? (Discussion #78)
Date: 28/12/23 03:47:47

Hi!
This is actually the intended behavior. The search view is to search
for individual songs only. Choosing "AlbumArtist" on the left simply
has the effect that only songs are shown where the "AlbumArtist" tag
contains the search request. It is planned to support a search/filter
functionality for artists in the future. Take a look at #86 for
reference.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID:
***@***.***>

**Reply by SoongNoonien** [reply]
> would you prefer comments for "Master" to
be raised on the issue tracker or kept separate in discussions /
emailed ?

If you found an actual bug please open a real bug report. Otherwise a discussing is also fine.
